src/VisualStudio/Core/Def/Implementation/SemanticClassificationCache/VisualStudioSemanticClassificationCacheService.cs

```diff
@@ -52,18 +52,16 @@ internal class VisualStudioSemanticClassificationCacheService
                 return default;
             }
 
-            var classifiedSpans = await client.RunRemoteAsync<SerializableClassifiedSpans>(
-                WellKnownServiceHubService.CodeAnalysis,
-                nameof(IRemoteSemanticClassificationCacheService.GetCachedSemanticClassificationsAsync),
-                solution: null,
-                arguments: new object[] { documentKey.Dehydrate(), textSpan, checksum },
+            var classifiedSpans = await client.TryInvokeAsync<IRemoteSemanticClassificationCacheService, SerializableClassifiedSpans?>(
+                (service, cancellationToken) => service.GetCachedSemanticClassificationsAsync(documentKey.Dehydrate(), textSpan, checksum, cancellationToken),
                 callbackTarget: null,
                 cancellationToken).ConfigureAwait(false);
-            if (classifiedSpans == null)
+
+            if (!classifiedSpans.HasValue || classifiedSpans.Value == null)
```

Comment:
i'm a bit confused by this.  is it a nullable that wraps a non-nullable?

Reply:
It's `Optional<T?>` where T : class.

Reply:
`default(Optional<T>)` means the remote call failed and gold-bar was displayed.
`null` means we don't have any spans (or whatever the service semantics is)